portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,30 +44,31 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
+        Integer numComments = Integer.parseInt(request.getParameter("num-comments"));
+
         List<Comment> comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
-        long id = entity.getKey().getId();
-        String text = (String) entity.getProperty("text");
-        long timestamp = (long) entity.getProperty("time");
-
-        Comment comment = new Comment(id, text, timestamp);
-        comments.add(comment);
-    }
+            if (numComments > 0) {
```

Comment:
I don't know if this is possible, but can you limit the number of results returned when constructing the query above?

Reply:
I'm not quite sure what you mean by that. When the get request is sent to this servlet, I have it set up so that only the specified number of comments will be added to an ArrayList and then returned (as long as numComments is greater than 0, additional comments will be added to the resulting list). Is there something different from what I've done that you're thinking of?

Reply:
Right now, you're running the query and then picking out the first numComments results.  I was just curious whether it was possible to specify such a limit in the query itself, so that "results" only contains at most numComments entries when you call datastore.prepare(query).

Looks like it is possible if you append a ".Limit(numComments)" to line 42 above: https://cloud.google.com/datastore/docs/concepts/queries#cursors_limits_and_offsets